src/PandoraMonitoring.cc

```diff
@@ -783,15 +838,43 @@ TEveElement *PandoraMonitoring::VisualizeVertices(const pandora::VertexList *con
     TEvePointSet *pTEvePointSet = new TEvePointSet(name.c_str(), 1);
     pTEvePointSet->SetOwnIds(kTRUE);
 
-    for (VertexList::const_iterator iter = pVertexList->begin(), iterEnd = pVertexList->end(); iter != iterEnd; ++iter)
+    bool isU(VERTEX_U == pVertexList->front()->GetVertexType());
```

Comment:
Need to know that pVertexList isn't empty, or front() will raise and exception.

Also, should note that VertexLists can contain vertices of different types. CaloHitLists can also contain CaloHits of different types, but LArTPC Pandora reco is configured to ensure that *Clusters* can only contain CaloHits of a single type, defined by the type of the first CaloHit added.

Obviously, for ILC, we typically expect Clusters with both e.g. ECAL and HCAL CaloHits.

Right now, for LArTPC, the only Vertices created are of type VERTEX_3D.

Reply:
Will add a try catch block before the loop to check the vertex list is not empty.